.cursor/rules/typescript.mdc

```diff
@@ -0,0 +1,37 @@
+---
+description:
+globs: *.ts, *.tsx
+alwaysApply: false
+---
+# TypeScript Guide
+
+- TypeScript version: 5.7.3
+- Linter: eslint
+- Formatter: prettier
+- Framework: React 18
+- Styling: @emotion/styled
+- Build tool: vite
+- Package manager: yarn v4 with workspaces
+
+## Key TypeScript Principles:
+
+- Prefer functional, declarative programming.
+- Prefer iteration and modularization over duplication.
+- Use descriptive variable names with auxiliary verbs (e.g., isLoading).
+- Use the Receive an Object, Return an Object (RORO) pattern.
+- Ensure functions have explicit return types.
+
+## Key Frontend Principles:
+
+- Leverage all of best practices of React 18.
+- Write performant frontend code.
+- Ensure referential stability by leveraging React Hooks.
+- Prefix event handlers with "handle" (eg: handleClick, handleSubmit).
+- Favor leveraging @emotion/styled instead of inline styles.
```

Comment:
should we specify that we (currently) use the [object style](https://emotion.sh/docs/object-styles) via emotion?